### PR TITLE
Clean up tasks & configurations, add shared testConfig

### PR DESCRIPTION
**Pull request by xerus2000**
This hopefully concludes the rampage in the last few days that ate up so many precious hours, trying to understand Gradle in depth.

One of the most important learnings & new features is a shared sourceset in the sdk, which via a configuration is depended upon in all projects to sync the KotestConfig and make util classes (currently only XStreamMatcher) available without exposing internals.
Other than that there are a few cleanups and fixes, nothing ground-breaking but generally improved cleanliness. @SKoschnicke should especially check the docker changes (https://github.com/CAU-Kiel-Tech-Inf/backend/commit/7a93580531c8f3e5d1dcdf4edc0a58dbb9817e27).

Some related reading material:
-   add compilation output to configuration via [artifact](https://docs.gradle.org/current/userguide/cross_project_publications.html#sec:simple-sharing-artifacts-between-projects) from compileKotlin task outputs
-   [Configurations & dependencies](https://docs.gradle.org/current/userguide/declaring_dependencies.html)
-   [Gradle standard configurations overview](https://docs.gradle.org/current/userguide/java_plugin.html#tab:configurations)

### CODE REVIEW

File: server/build.gradle.kts [hunk 1]
```diff
@@ -60,25 +63,28 @@ tasks {
...
     
     val dockerImage by creating(Exec::class) {
+        group = "application"
         dependsOn(makeRunnable)
+        workingDir = projectDir.resolve("configuration")
 		doFirst {
 			val tag = Runtime.getRuntime().exec(arrayOf("git", "rev-parse", "--short", "--verify", "HEAD")).inputStream.reader().readText().trim()
-			val relativeRunnable = runnableDir.relativeTo(project.projectDir)
+			val relativeRunnable = runnableDir.relativeTo(workingDir)
 			commandLine("docker", "build", "--no-cache", "-t", "swc_game-server:latest", "-t", "swc_game-server:$tag", "--build-arg", "game_server_dir=$relativeRunnable", ".")
```
Comment (SKoschnicke) [inline server/build.gradle.kts:79]: This won't work with the `relativeRunnable` path `/server/build/runnable` being outside the working dir `/server/configuration`. All resources for the container have to be under the working directory which is specified by the last argument of the `docker build` command (currently `.`). You can copy `/server/build/runnable` into the working dir or give docker another working directory and change the paths to the other resources in `Dockerfile`. I would prefer the first option.

```
> Task :server:dockerImage FAILED
Sending build context to Docker daemon  11.78kB

Step 1/9 : FROM openjdk:8-jre
 ---> 39f48bacf78c
Step 2/9 : ARG game_server_dir
 ---> Running in 4527c39a980f
Removing intermediate container 4527c39a980f
 ---> de29cfdb46e1
Step 3/9 : ADD $game_server_dir /app
ADD failed: Forbidden path outside the build context: ../build/runnable ()

FAILURE: Build failed with an exception.
```

**Review by SKoschnicke — CHANGES_REQUESTED** [review]

File: server/build.gradle.kts (hunk 1 above)
Comment (xerus2000) [reply server/build.gradle.kts:82]: please check the new version

Comment (SKoschnicke) [reply server/build.gradle.kts:82]: works with my additional changes :+1:

**Review by SKoschnicke — APPROVED** [review]
LGTM!

**Pull request merged**